src/cls/rbd/cls_rbd.h

```diff
@@ -359,8 +359,17 @@ struct cls_rbd_rwlcache_map {
     }
   };
 
+  // cache allocated but not receive ack in uncommitted_caches
+  std::map<epoch_t, std::pair<utime_t, struct Cache>> uncommitted_caches;
+
+  // uncommitted_cache recive ack in time and result is ok. Move cahe from uncommitted_cache
+  // to caches.
   std::map<epoch_t, struct Cache> caches;
 
+  // Primary can't free related daemon's cache-file, etc rdma-disconnect.
+  // So cache move uncommitted_caches or caches to free_daemon_space_caches;
+  std::map<epoch_t, struct Cache> free_daemon_space_caches;
+
```

Comment:
Is it better which  move uncommitted_caches /caches/free_daemon_space_caches into first patch and based on to refactor code? @hualongfeng

Reply:
done